src/org/rascalmpl/shell/REPLRunner.java

```diff
@@ -50,6 +49,11 @@ private static ILanguageProtocol makeInterpreter(InputStream stdin, OutputStream
             new RascalInterpreterREPL(prettyPrompt, allowColors, getHistoryFile()) {
                 @Override
                 protected Evaluator constructEvaluator(InputStream input, OutputStream stdout, OutputStream stderr, IDEServices services) {
+                    if (services instanceof BasicIDEServices) {
+                        // TODO: find a better way to promote this wrapper to stdout
```

Comment:
agreed, that would be nice.

Reply:
This code has been removed by now. It was dead because the right object are now injected from the outside already.